### DISCUSSION TITLE
请问如何在MSVC上编译

### DISCUSSION THREAD

**Discussion by weboose** [Tencent/ScriptX#59] [Q&A] [answered]
尝试使用msvc2017进行编译无法通过

**Comment by yqs112358** [comment, marked as answer]
你可以先用cmake编译成静态库，然后链接到VS项目里面去
静态库生成样例：https://github.com/LiteLDev/LiteXLoader/tree/main/ScriptX

**Reply by weboose** [reply]
我确实是想编译成静态库进行使用，但是编译无法通过，在这里报告语法错误
![QW}}GTIT{_}8 VC{15E`ILY](https://user-images.githubusercontent.com/20901213/143674787-cbd4d84c-1e8e-4f6a-9696-777a60c9e7e9.png)

**Reply by LanderlYoung** [reply]
cmake是正解，具体是什么语法错误？先本地改一下看看吧

**Reply by weboose** [reply]
就是这个报错
![image](https://user-images.githubusercontent.com/20901213/143802911-5266f89b-0617-4bd0-9345-eee4f105d6ea.png)
![image](https://user-images.githubusercontent.com/20901213/143802929-e53d7847-8703-4fc3-bd92-25d013fafb65.png)

**Reply by LanderlYoung** [reply]
改成 `mq->postMessage(msg)` 试试看，可能是编译器的兼容问题

**Reply by weboose** [reply]
改为```mq->postMessage(msg)```确实可以，这样是否和```mq->template postMessage(msg)```等效呢？

**Reply by LanderlYoung** [reply]
功能是等效的，你可以先这样跑起来。稍后我具体check一下

**Reply by weboose** [reply]
好的，谢谢，现在我这样是可以编译通过了，使用也一切正常，我所使用的编译器是VS 2017 64位编译。

**Reply by LanderlYoung** [reply]
fixed in https://github.com/Tencent/ScriptX/pull/63
